### DISCUSSION TITLE
Computation of Local Angle of Attack

### DISCUSSION THREAD

**Discussion by tchatte3** [OpenFAST/openfast#1438] [General]
Hello OpenFAST Team,

I was trying look -- how is the local angle of attack computed in OpenFAST `Aerodyn` kernel [for steady model]?
Is it `tan(AoA) = (U_axial - x_dot)/(omega(t)*r - U_\theta - y_dot)` ?
How does the definition change for yawed turbine cases?
Is it ``tan(AoA) = (U_axial*cos(\gamma) - x_dot)/(omega(t)*r - U_\theta - y_dot)``, where \gamma is the yaw-misalignment?
I am considering `x_dot` and `y_dot` as the blade vibrational velocities interpolated to Aerodyn mesh (atleast in OpenFAST versions 2.X)

Any insights regarding this will be really helpful.
Best,
Tanmoy

**Comment by ebranlard** [comment]
Hi @tchatte3, 

Your equation for the angle of attack resembles the equation for the flow angle (`phi`), so I would assume that is the case. In our coordinate system, we can assume that:
 `alpha = phi-beta` 
where beta is the pitch and twist angle. In the general case, where the blade is coned, yawed, or deflected, the coordinate system of the airfoil section can be anywhere in space, and therefore, the notation "axial" and "theta" might not necessarily apply (though I understand what you mean by them). In AeroDyn, the blade element part (where the angle of attack is defined) and the momentum part (where the inductions are obtained) are currently done in a special coordinate system in which there is no pitch, no twist, and no sweep compared to the blade-root coordinate system. (We will change that in the near future). I'll write "x" and "y" this coordinate system. For a straight blade, nonyawed, "x" is "axial", "y" is "theta" (or -"theta" depending on your convention). 

Your equation does not explicitly mentions induced velocities (you probably assume that they are in `U_axial` and `U_theta`). In AeroDyn, the elastic velocities and the wind velocities are gathered together as a common flow velocity, that I will note "V". Typically :
**`V = Vwind -Velast + dVdist`** 
where **`Vwind`** is the undisturbed wind vector (including shear, turbulence), **`Velast`** is the velocity of the section (`x_dot` and `y_dot` in your post), and **`dVdist`** is the disturbance from the tower. In AeroDyn, this flow velocity `V` is used to scale the induced velocities. Other implementations indeed use an approach similar to what you mention and have the "elastic" velocity as a separate contribution in the velocity triangle and scaled the induced velocity with the wind speed only. 

In AeroDyn, the equation for the flow angle would be something like: 
`tan(phi) = Vx(1-a)   /   Vy(1+a')`
Where `Vx` and `Vy` are the project components of the vector **`V`** mentioned above. 
In the yawed case, a similar equation will be used, the wind velocities are projected into the section coordinate system (which is yawed), and there is an additional correction on the induction factors. It would probably be too complex to detail this here. 

You can also look at the following discussion to understand the AeroDyn outputs:
https://github.com/OpenFAST/openfast/discussions/1028


I hope that helps a bit!

**Comment by tchatte3** [comment]
Hello @ebranlard,
Thank you for the detailed response. 
Yes, I rmissed the `\beta`  part -- you are right, since I was more concerned with the definition of `phi` based on the velocities. 
Also, you are right -- since I am formulating this more like an actuator-line model `V_axial` takes care of the axial induction factor and `V_\theta` is accounting for the tangential induction factor. 

But regarding the yawed question -- let me simplify the question a little bit. 
If we imagine we have a rigid blades V_elast = 0, and tower disturbance is zero: V_dist = 0.
How does the equation of flow angle change:

No yaw:
`tan(phi) = V_axial/(\omega(t)*R - V_\theta) = Vx(1-a)/Vy(1+a')`
Yaw:
`tan(phi) = V_axial cos(\gamma)/(\omega(t)*R  - V_\theta) = Vx cos(\gamma) (1 + a_mod)/Vy(1 + a'_mod)` 
V_x cos(\gamma) should be essentially the induced velocity projected in the yawed system. Apologies for moving from your formulation as I am still trying to understand how openFAST is trying to implement the angle of attack formulation. You mentioned about it in #1028. But, I am curious if the `V_x sin(\gamma)` term is used to model any `radial aero-drag` model in Aerodyn. 

Best,
Tanmoy

**Comment by ebranlard** [comment]
Hi Tanmoy,

For a straight blade (no prebend, no sweep, no cone), under a yaw angle `gamma`, using OpenFAST blade root coordinate system, we have:
`Vund,x  = V0 * cos(gamma) `
`Vund,y = -V0 * sin(gamma) * cos(psi)`
`Vstr,x  = 0 `
`Vstr,y  = - Omega r `
where `psi` is the azimuthal blade position (0 for the blade pointing up), `Omega` is the rotor speed, and `r` the (orthogonal) distance to the hub. 
If we neglect the contribution from the tower (`Delta Vdist`), this leads to:
`Vx =  V0 * cos(gamma)`
`Vy =  -V0 * sin(gamma) * cos(psi)  + Omega r`
With these notations, the equation for the flow angle still has the same form:
`tan(phi) = Vx(1-a_mod)  /  Vy(1+a'_mod)` 
The modified axial and tangential induction factors are obtained based on the skew correction. 

Does that help?

**Reply by tchatte3** [reply]
@ebranlard This surely helps Emmanuel. Thanks for taking time to explain this. 

Best,
Tanmoy
